src/throttle/index.ts

```diff
@@ -19,26 +29,43 @@ export function throttle<T>({
   target = createEvent<T>(),
 }: {
   source: Unit<T>;
-  timeout: number;
+  timeout: number | Store<number>;
   name?: string;
   target?: Unit<any>;
 }): EventAsReturnType<T> {
   if (!is.unit(source)) throw new TypeError('source must be unit from effector');
 
-  if (typeof timeout !== 'number' || timeout < 0)
-    throw new Error('timeout must be positive number or zero');
+  const $timeout = toStoreNumber(timeout);
 
-  const timerFx = createEffect(
-    () => new Promise((resolve) => setTimeout(resolve, timeout)),
+  const timerFx = createEffect<number, void>(
+    (timeout) => new Promise((resolve) => setTimeout(resolve, timeout)),
   );
 
-  guard({
-    source,
-    filter: timerFx.pending.map((pending) => !pending),
+  sample({
+    source: $timeout,
+    clock: guard({
+      source,
+      filter: timerFx.pending.map((pending) => !pending),
+    }),
     target: timerFx,
   });
 
   sample({ source, clock: timerFx.done, target });
 
   return target as any;
 }
+
+function toStoreNumber(value: number | Store<number> | unknown): Store<number> {
+  if (is.store(value)) return value;
+  if (typeof value === 'number') {
+    if (value < 0 || !Number.isFinite(value))
+      throw new Error(
+        `timeout must be positive number or zero. Received: "${value}"`,
+      );
+    return createStore(value, { name: '$timeout' });
+  }
+
+  throw new TypeError(
+    `timeout parameter in interval method should be number or Store. "${typeof value}" was passed`,
```

Comment:
I think, there is a mistake - wrong method name 🤔 

Also, i think, that it can be not mention at all, since stacktrace of error will lead to method call anyway
```suggestion
    `timeout parameter should be number or Store. "${typeof value}" was passed`,
```